src/hdx_cli_toolkit/utilities.py

```diff
@@ -335,3 +335,31 @@ def print_dictionary_comparison(
                 flush=True,
             )
     print("-" * total_width, flush=True)
+
+
+def query_dict(
+    key_: str,
+    output: list[dict[str, Any]],
+    dataset_dict: dict[str, Any],
+    output_row: dict[str, Any],
+) -> list[dict[str, Any]]:
+    if "." not in key_:
+        output_row[key_] = dataset_dict.get(key_, "Key absent")
+        output.append(output_row)
+    else:
+        key1, key2 = key_.split(".")
+        intermediate_value = dataset_dict.get(key1, "Key1 absent")
+        if isinstance(intermediate_value, dict):
+            output_row[key_] = intermediate_value.get(key2, "Key2 absent")
+            output.append(output_row)
+        elif isinstance(intermediate_value, list):
+            for item in intermediate_value:
+                tmp_row = output_row.copy()

```

Comment:
Why do you create a copy of output_row here?

Reply:
I've been burnt before by the mutability of dictionaries!